server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -86,15 +88,15 @@ protected RestRequest(NamedXContentRegistry xContentRegistry, Map<String, String
 
     private RestRequest(NamedXContentRegistry xContentRegistry, Map<String, String> params, String path,
                         Map<String, List<String>> headers, HttpRequest httpRequest, HttpChannel httpChannel, long requestId) {
-        final XContentType xContentType;
-        try {
-            xContentType = parseContentType(headers.get("Content-Type"));
-        } catch (final IllegalArgumentException e) {
+        try{
+            this.parsedAccept = parseHeaderWithMediaType(httpRequest.getHeaders(), "Accept");
+            this.parsedContentType = parseHeaderWithMediaType(httpRequest.getHeaders(), "Content-Type");
+            if (parsedContentType != null) {
+                this.xContentType.set(parsedContentType.toMediaType(XContentType.MEDIA_TYPE_REGISTRY));
+            }
+        }catch (IllegalArgumentException e){
             throw new ContentTypeHeaderException(e);
```

Comment:
Since this can now be either Accept or Content-Type we probably want to change this to `MediaTypeHeaderException`